packages/payload/src/queues/operations/runJob/getUpdateJobFunction.ts

```diff
@@ -0,0 +1,23 @@
+import type { PayloadRequest } from '../../../types/index.js'
+import type { BaseJob } from '../../config/types/workflowTypes.js'
+
+export type UpdateJobFunction = (jobData: Partial<BaseJob>) => Promise<BaseJob>
+
+export function getUpdateJobFunction(job: BaseJob, req: PayloadRequest): UpdateJobFunction {
```

Comment:
i think this whole thing needs to go, we should not be allowing the mutation of jobs within task / workflow runners

Reply:
I kept it so that updates from payload are reflected in the job object - I did ensure that mutations of the job object do not make it back to payload though